crowdsourcing/viewsets/project.py

```diff
@@ -70,102 +70,192 @@ def list(self, request, *args, **kwargs):
+                                                                INNER JOIN (
+                                                                    SELECT task_id, id
+                                                                    FROM crowdsourcing_taskworker
+                                                                    GROUP BY task_id, id
+                                                                ) tw
+                                                                ON tw.id = task_worker_id
+                                                                GROUP BY task_worker_id, task_id
+                                                            ) tw_by_task
+                                                        GROUP BY task_id
+                                                        ) submit_by_task
+                                                        ON t.id=task_id
+                                                    ) submit_with_module
+                                                    GROUP BY module_id
+                                                ) task_count_by_module
+                                                ON task_count_by_module.module_id = m.id
+                                                INNER JOIN (
+                                                    SELECT m.id, (m.repetition * m.task_time * 60 * COUNT(t.id) / num_active_workers) AS module_length
+                                                    FROM crowdsourcing_module m
+                                                    INNER JOIN crowdsourcing_task t ON t.module_id=m.id
+                                                    INNER JOIN (
+                                                        SELECT COUNT(*) AS num_active_workers
+                                                        FROM (
+                                                            SELECT w.id, EXTRACT('EPOCH' FROM NOW() - u.last_login) AS elapsed
+                                                            FROM crowdsourcing_worker w
+                                                            INNER JOIN crowdsourcing_userprofile up ON profile_id=up.id
+                                                            INNER JOIN auth_user u ON u.id = up.user_id
+                                                        ) worker_last_login
+                                                        WHERE elapsed IS NOT NULL AND elapsed < EXTRACT('EPOCH' FROM INTERVAL '1 day')
+                                                    ) mod_task_workers
+                                                    ON TRUE
+                                                    GROUP BY m.id, num_active_workers
+                                                ) module_time
+                                                ON module_time.id=m.id
+                                            ) cascade_data
```

Comment:
thats why i multiply by 60 here so that it can be compared with elapsed_time which afaik is in seconds.